### PR TITLE
some more regex string fixes; and: use numpy.nan instead of numpy.NAN

### PR DESCRIPTION
**Pull request by hsk17**
numpy-2 has dropped NAN and NaN
(numpy documentation still says "NaN and NAN are aliases of nan" but this is obviously wrong)

### CODE REVIEW

File: gromacs/qsub.py [hunk 1]
```diff
@@ -430,15 +430,15 @@ def write_script(template):
                 template,
                 [
                     ("^ *DEFFNM=", "(?<==)(.*)", deffnm),
-                    ("^#.*(-N|job_name)", "((?<=-N\s)|(?<=job_name\s))\s*\w+", jobname),
+                    ("^#.*(-N|job_name)", r"((?<=-N\s)|(?<=job_name\s))\s*\w+", jobname),
```
Comment (orbeckst) [inline gromacs/qsub.py:433]:
```suggestion
                    (
                       "^#.*(-N|job_name)",
                       r"((?<=-N\s)|(?<=job_name\s))\s*\w+",
                       jobname,
                    ),
```

**Review by orbeckst — APPROVED** [review]
Thank you for the fix. Sorry, I hadn't seen the PR.

I'll also add you to the CHANGES.

File: AUTHORS [hunk 2]
```diff
@@ -68,6 +68,11 @@ year of their first commit. GitHub handle is optional.
 ----
 
 - Jinzhe Zeng (@njzjz)
+
+2024
+----
+
+-  @hsk17
```
Comment (orbeckst) [inline AUTHORS:75]: @hsk17 please add your full name if you want to be acknowledged by it

**Review by orbeckst — APPROVED** [review]
I will merge the PR in a few days or sooner if I hear from you. Thank you for your contribution.

File: AUTHORS (hunk 2 above)
Comment (hsk17) [reply AUTHORS:75]: Friedrich Haubensak

thanks :-)

Comment (hsk17) [reply AUTHORS:75]: BTW:  for the numpy.nan issue, there is also https://github.com/Becksteinlab/numkit/pull/32

Comment (orbeckst) [reply AUTHORS:75]: thanks for the heads up -- will check numkit afterwards, feel free to ping me with @orbeckst to get my attention

**Pull request merged**